libcontainer/container_linux.go

```diff
@@ -775,6 +775,27 @@ func (c *linuxContainer) updateState(process parentProcess) error {
 	return json.NewEncoder(f).Encode(state)
 }
 
+func (c *linuxContainer) checkFreezer() (Status, error) {
+	path := c.cgroupManager.GetPaths()["freezer"]
+	if _, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			// Freezer subsystem is not available,Application is neither Checkpointed nor Destroyed
+			// return the status as Running
+			return Running, nil
+		}
+	}
+	contents, err := ioutil.ReadFile(filepath.Join(path, "freezer.state"))
+	if err != nil {
+		return 0, newSystemError(err)
+	}
+	freezerstate := string(contents)
+	if strings.TrimSpace(freezerstate) == "FROZEN" {
```

Comment:
On Mon, Sep 07, 2015 at 12:01:35AM -0700, Qiang Huang wrote:
> Maybe we should check both `FROZEN` and `FREEZING` here, we have
> `pausing` defined but never used, it should be returned here.

Previous thoughts on model state settings in [1,2,3].  But if we
already have state constants for each of Linux's three settings, I
think it makes sense to use them (at least until the runtime-agnostic
spec has something to say about freezing).

[1]: https://github.com/opencontainers/runc/pull/218#issuecomment-133487383
[2]: https://github.com/opencontainers/runc/pull/218#issuecomment-133503694
[3]: https://github.com/opencontainers/runc/pull/218#issuecomment-133520061